src/App.tsx

```diff
@@ -1,61 +1,128 @@
 import './App.scss';
+import { FormEvent, useState } from 'react';
+import { TodoList } from './components/TodoList/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [userName, setUserName] = useState('');
+  const [isTitleEmpty, setIsTitleEmpty] = useState(false);
+  const [isUserSelected, setIsUserSelected] = useState(false);
+
+  const retrieveUser = (): Todo[] => {
+    return todosFromServer.map(todo => {
+      const user = usersFromServer
+        .find(serverUser => serverUser.id === todo.userId) || null;
+
+      return { ...todo, user };
+    });
+  };
+
+  const retrievedUser = retrieveUser();
+  const [todos, setTodos] = useState(retrievedUser);
+
+  const handleTitleChange = (event: FormEvent<HTMLInputElement>) => {
+    setTitle(event.currentTarget.value);
+    setIsTitleEmpty(false);
+  };
+
+  const handleUserChange = (event: FormEvent<HTMLSelectElement>) => {
+    setUserName(event.currentTarget.value);
+    setIsUserSelected(false);
+  };
+
+  const addTodo = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const selectedUser = usersFromServer.find(user => user.name === userName);
+    const todosIds = todos.map(todo => todo.id);
+    const maxId = Math.max(...todosIds);
+    const trimedTitle = title.trim();
+
+    if (title.length < 1 || trimedTitle.length < 1) {
+      setIsTitleEmpty(true);
+    }
+
+    if (!selectedUser) {
+      setIsUserSelected(true);
+    }
+
+    if (
+      !isTitleEmpty
+      && !isUserSelected
+      && trimedTitle
+      && selectedUser
+    ) {
+      setTodos([
+        ...todos,
+        {
+          id: maxId + 1,
+          userId: selectedUser?.id,
+          title: trimedTitle,
+          completed: false,
+          user: selectedUser || null,
+        }]);
+      setTitle('');
+      setUserName('');
+    }
+  };
```

Comment:
```suggestion
      if (!selectedUser || ! trimedTitle) {
        return;
      }
      
      setTodos([...todos, {
          id: maxId + 1,
          userId: selectedUser?.id,
          title: trimedTitle,
          completed: false,
          user: selectedUser || null,
        }]);
        
      setTitle('');
      setUserName('');
  };
```

Reply:
if i take setTodos out of IF block i have to  make changes in type Todo.ts. Is that okay?

Reply:
also if i do the if checks `if (!selectedUser || ! trimedTitle) {
        return;
      }` im able to add title without user checked. I know my checks looked ugly but they did work correctly 😅